File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& s) {
        unordered_set<int> ht;
        
        for(auto ele : s) ht.insert(ele);
        int longest = 0;
        for(int i = 0; i < s.size(); i++){
            if(ht.find(s[i]-1) == ht.end()){
                int length = 0;
                while(ht.find(s[i] + length) != ht.end()){
                    length++;
                }
                longest = max(length, longest);
            }
        }
        return longest;
    }
};

File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.py
class Solution(object):
    def longestConsecutive(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        ht = set(nums)
        count = 0
        for i in nums:
            if i-1 not in ht: 
                length = 0
                while(i+length in ht):
                    length += 1
                count = max(count, length)
        return count